sql/core/src/test/scala/org/apache/spark/sql/JoinHintSuite.scala

```diff
@@ -695,11 +695,11 @@ class JoinHintSuite extends PlanTest with SharedSparkSession with AdaptiveSparkP
     val hintAppender = new LogAppender(s"join hint check for equi-join")
     withLogAppender(hintAppender, level = Some(Level.WARN)) {
       assertBroadcastNLJoin(
-        df1.hint("SHUFFLE_HASH").join(df2, $"a1" !== $"b1"), BuildRight)
+        df1.hint("SHUFFLE_HASH").join(df2, $"a1"=!= $"b1"), BuildRight)
     }
     withLogAppender(hintAppender, level = Some(Level.WARN)) {
       assertBroadcastNLJoin(
-        df1.join(df2.hint("MERGE"), $"a1" !== $"b1"), BuildRight)
+        df1.join(df2.hint("MERGE"), $"a1" =!= $"b1"), BuildRight)
```

Comment:
Is this the last instance in Spark repo?

Reply:
![image](https://github.com/apache/spark/assets/1475305/0e7caa02-ba4c-4e01-8a76-0efb4ba37993)

Yes, other calls are in the DeprecatedAPISuite, we should keep them.